src/client.rs

```diff
@@ -2255,4 +2264,85 @@ mod test {
             "me/player/shuffle?state=true&device_id=fdafdsadfa"
         );
     }
+
+    #[test]
+    fn test_cartesian_product() {
+        let attributes = [
+            "acousticness",
+            "danceability",
+            "duration_ms",
+            "energy",
+            "instrumentalness",
+            "key",
+            "liveness",
+            "loudness",
+            "mode",
+            "popularity",
+            "speechiness",
+            "tempo",
+            "time_signature",
+            "valence",
+        ];
+        let prefixes = ["min", "max", "target"];
+
+        let iterator_product_result = attributes
+            .iter()
+            .flat_map(|attribute| {
+                prefixes
+                    .iter()
+                    .map(move |prefix| format!("{}_{}", prefix, attribute))
+            })
+            .collect::<Vec<_>>();
+        let mut loop_product_result = vec![];
+        for attribute in attributes.iter() {
+            for prefix in prefixes.iter() {
+                let param = format!("{}_{}", prefix, attribute);
+                loop_product_result.push(param);
+            }
+        }
+        assert!(iterator_product_result
+            .iter()
+            .eq(loop_product_result.iter()));
+    }
+    #[test]
+    fn test_cartesian_product_and_filter_map() {
```

Comment:
is this test really necessary?

Reply:
It's not that necessary, I will remove it.

Reply:
fixed